client.go

```diff
@@ -197,7 +203,11 @@ func (c *Client) registerProviders() {
 	driverAMT := intelamt.New(c.Auth.Host, c.Auth.User, c.Auth.Pass, iamtOpts...)
 	c.Registry.Register(intelamt.ProviderName, intelamt.ProviderProtocol, intelamt.Features, nil, driverAMT)
 
-	// register Dell gofish provider
+	return nil
+}
+
+// register Dell gofish provider
+func (c *Client) registerDellProvider() error {
```

Comment:
this func will never return an error. i would suggest not having any return value.